### DISCUSSION TITLE
Is there an alternative to the `vscode` module?

### DISCUSSION THREAD

**Discussion by terrymun** [microsoft/vscode-discussions#816] [Extension Development QnA] [answered]
The `vscode` module is marked as deprecated and to be replaced by `@types/vscode` and `vscode-test`: https://www.npmjs.com/package/vscode

However `vscode-test` does not contain all the API that `vscode` offers. What would be the solution? For example, there is no equivalent of `vscode.commands.registerCommand`.

**Comment by HatemMn** [comment, marked as answer]
> For example, there is no equivalent of vscode.commands.subscriptions.

Are you talking about ` vscode.context.subscriptions` ?

Also, did you include vscode as an engine in the extension manifest ?

```
  "engines": {
    "vscode": "^1.78.0"
  },
```

**Reply by terrymun** [reply]
@HatemMn Oh sorry, my bad when copying and pasting. Based on the docs for VSCode, commands can be registered as such:

```ts
import { commands } from 'vscode';

export const activate = (context: ExtensionContext) => {
  context.subscriptions.push(
    commands.registerCommand('my-custom-plugin.someCommand', someCommand)
  );
}

export const deactivate = () => {};
```

I am at lost at what should I use instead of `import { commands } from 'vscode'` sine the package is already marked for deprecation.

p/s: Yes, I have `vscode` included as an engine in the extension manifest.

**Reply by terrymun** [reply]
Update: it turns out that as long as the engine is specified in the manifest, I can safely delete `vscode` dependency from my package.json file :) 

The errors I'm getting are purely from TypeScript and installing `@types/vscode` helped.

Thanks for the help!
